src/components/Plugin/Hal/CustomBlock.vue

```diff
@@ -0,0 +1,55 @@
+<script setup>
+const props = defineProps({
+  space: Object
+});
+
+const getSpaceId = () => {
+  const spaceId = props.space.id;
+  return spaceId;
+};
+
+const getSpaceName = () => {
+  const spaceName = props.space.name;
+  return spaceName;
+};
```

Comment:
Also here I would just use `props.space.name` inside script and `space.name` inside template.